deebotozmo/cli.py

```diff
@@ -0,0 +1,339 @@
+    config['verify_ssl'] = verify_ssl
+    write_config(config)
+    click.echo("Config saved.")
+    sys.exit(0)
+
+@cli.command(help='Play welcome sound')
+@coro
+async def playsound():
+    vacbot = DoLogin()
+    await vacbot.run()
+    await vacbot.bot.execute_command(PlaySound())
+    await vacbot.goodbye()
+
+@cli.command(help='Auto clean')
+@coro
+async def clean():
+    vacbot = DoLogin()
+    await vacbot.run()
+    await vacbot.bot.execute_command(CleanStart())
+    await vacbot.goodbye()
+
+@cli.command(help='Cleans provided area(s), ex: "-602,1812,800,723"', context_settings={"ignore_unknown_options": True})
+@click.argument('area', type=click.STRING, required=True)
+@click.argument('cleanings', type=click.STRING, required=False)
+@coro
+async def CustomArea(area, cleanings=1):
+    vacbot = DoLogin()
+    await vacbot.run()
+    await vacbot.bot.execute_command(CleanCustomArea(map_position=area, cleanings=cleanings))
+    await vacbot.goodbye()
+
+
+@cli.command(help='Cleans provided rooms(s), ex: "0,1" | Use GetRooms to see saved numbers',
+             context_settings={"ignore_unknown_options": True})
+@click.argument('rooms', type=click.STRING, required=True)
+@click.argument('cleanings', type=click.STRING)
+@coro
+async def SpotArea(rooms, cleanings=1):
+    vacbot = DoLogin()
+    await vacbot.run()
+    await vacbot.bot.execute_command(CleanSpotArea(rooms, cleanings))
+    await vacbot.goodbye()
+
+
+@cli.command(help='Set Clean Speed')
+@click.argument('speed', type=click.STRING, required=True)
+@coro
+async def setfanspeed(speed):
+    vacbot = DoLogin()
+    await vacbot.run()
+    await vacbot.bot.execute_command(SetFanSpeed(speed))
+    await vacbot.goodbye()
+
+
+@cli.command(help='Set Water Level')
+@click.argument('level', type=click.STRING, required=True)
+@coro
+async def setwaterLevel(level):
+    vacbot = DoLogin()
+    await vacbot.run()
+    await vacbot.bot.execute_command(SetWaterLevel(level))
+    await vacbot.goodbye()
+
+
+@cli.command(help='Returns to charger')
+@coro
+async def charge():
+    vacbot = DoLogin()
+    await vacbot.run()
+    await vacbot.bot.execute_command(Charge())
+    await vacbot.goodbye()
+
+
+@cli.command(help='Pause the robot')
+@coro
+async def pause():
+    vacbot = DoLogin()
+    await vacbot.run()
+    await vacbot.bot.execute_command(CleanPause())
+    await vacbot.goodbye()
+
+@cli.command(help='Resume the robot')
+@coro
+async def resume():
+    vacbot = DoLogin()
+    await vacbot.run()
+    await vacbot.bot.execute_command(CleanResume())
+    await vacbot.goodbye()
+
+@cli.command(help='Get Clean Logs')
+@coro
+async def getCleanLogs():
+    vacbot = DoLogin()
+    await vacbot.run()
+
+    lock = asyncio.Lock()
+    await lock.acquire()
+    async def on_clean_event(event: CleanLogEvent):
+        print(json.dumps(asdict(event)))
+        lock.release()
+
+    listener = vacbot.bot.cleanLogsEvents.subscribe(on_clean_event)
+    vacbot.bot.cleanLogsEvents.request_refresh()
```

Comment:
```suggestion
```

Please remove the code line as it is not required and with this line you will fetch the data **twice**!!

Please also remove the other `request_refresh` code lines as I requested in the first review

Reply:
I had instances where I missed these and never got any data, I'll try again